src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,124 @@
+            FileWriter fw = new FileWriter(file);
+            fw.write("");
+            for (String s: data) {
+                fw.append(s);
+            }
+            fw.close();
+        } catch (IOException err) {
+            throw new DukeException("File Path is a directory -OR- Can't create file at location");
+        }
+    }
+
+    /**
+     * Read and interpret the saved file.
+     * @return ArrayList of Task that is saved inside the designated file. If file / directory does not exist, create.
+     * @throws DukeException If file is not found.
+     */
+    public ArrayList<Task> loadFile() throws DukeException {
+        try {
+            ArrayList<Task> loadedTask = new ArrayList<>();
+            Scanner scanner = new Scanner(this.file);
+            String[] dataRead;
+            while (scanner.hasNext()) {
+                dataRead = readSavedData(scanner.nextLine());
+                Task newTask = createSavedTask(dataRead);
+                loadedTask.add(newTask);
+            }
+            return loadedTask;
+        } catch (FileNotFoundException err) {
+            String[] fileDirectory = this.filePath.split("/");
+            String parentDirectory = "";
+            for (int i = 0; i < fileDirectory.length - 1; i++) {
+                parentDirectory += fileDirectory[i] + "/";
+            }
+            File f = new File(parentDirectory);
+            f.mkdirs();
+            throw new DukeException("Can't Find Save File. Starting New");
+        }
+    }
+
+    /**
+     * Split individual saved line into a 3 parts: Command, What, When
+     * @param inputLine A line from save file to be interpreted.
+     * @return A String array with 2 or 3 parts.
+     */
+    private String[] readSavedData(String inputLine) {
+        String[] parts = inputLine.split(" \\| ");
+        ArrayList<String> result = new ArrayList<>();
+        for (String part : parts) {
+            result.add(part.trim());
+        }
+        return result.toArray(parts);
+    }
+
+    /**
+     * Create a new Task based on the input.
+     * @param args The 2 or 3 parts command (Command, What, When).
+     * @return Task created based on the command
+     * @throws DukeException If the command requires (What, When) parts but one or more parts is missing.
+     */
+    private Task createSavedTask(String[] args) throws DukeException {
+        Task newTask;
+        try {
+            switch(args[0]) {
+            case "D":
+                newTask = new Deadline(args[2], args[3]);
+                break;
+            case "T":
+                newTask = new ToDo(args[2]);
+                break;
+            case "E":
+                newTask = new Event(args[2], args[3]);
+                break;
+            default:
+                throw new DukeException("Error: Saved File is badly corrupted");
+            }
+        } catch (Exception err) {
+            throw new DukeException("Error: Saved File is badly corrupted");
+        }
```

Comment:
Im confused about this part. You are throwing it, to catch it to throw it again without change? unless the constructors throw a different kind of Exception?